ADAL/src/utils/ios/ADKeychainUtil.m

```diff
@@ -53,15 +53,16 @@ + (NSString*)retrieveTeamIDFromKeychain:(ADAuthenticationError * __autoreleasing
     NSDictionary *query = @{ (id)kSecClass : (id)kSecClassGenericPassword,
                              (id)kSecAttrAccount : @"teamIDHint",
                              (id)kSecAttrService : @"",
-                             (id)kSecAttrAccessible : (id)kSecAttrAccessibleAlways,
                              (id)kSecReturnAttributes : @YES };
     CFDictionaryRef result = nil;
     
     OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
     
     if (status == errSecItemNotFound)
     {
-        status = SecItemAdd((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
+        NSMutableDictionary* addQuery = [query mutableCopy];
+        [addQuery setObject:(id)kSecAttrAccessibleAlways forKey:(id)kSecAttrAccessible];
```

Comment:
This does not address the original bug where if the item is still around with the wrong accessibility attribute if the device was locked we wouldn't be able to get the keychain team ID.